lib/node/disseminator.ml

```diff
@@ -0,0 +1,38 @@
+type pool_elt = {
+  message : Message.t;
+  remaining : int;
+}
+
+module DigestSet = Set.Make(Digest)
+
+type t = {
```

Comment:
I feel like making everything `mutable` would be the same as making nothing `mutable`... Wouldn't it be better to return a new `Disseminator.t` instead of having all these mutable types?

Reply:
Handled